main/lsp/ExtractVariable.cc

```diff
@@ -186,16 +186,24 @@ class LocSearchWalk {
     void preTransformExpressionPtr(core::Context ctx, const ast::ExpressionPtr &tree) {
         if (tree.loc() == targetLoc.offsets()) {
             // It's not valid to extract the following node types
-            if (!ast::isa_tree<ast::Break>(tree) && !ast::isa_tree<ast::Next>(tree) &&
-                !ast::isa_tree<ast::Return>(tree) && !ast::isa_tree<ast::Retry>(tree) &&
-                !ast::isa_tree<ast::RescueCase>(tree) && !ast::isa_tree<ast::InsSeq>(tree)) {
-                matchingNode = &tree;
-                ENFORCE(!enclosingClassStack.empty());
-                matchingNodeEnclosingClass = enclosingClassStack.back();
-                if (!enclosingMethodStack.empty()) {
-                    matchingNodeEnclosingMethod = enclosingMethodStack.back();
+            if (ast::isa_tree<ast::Break>(tree) || ast::isa_tree<ast::Next>(tree) || ast::isa_tree<ast::Return>(tree) ||
+                ast::isa_tree<ast::Retry>(tree) || ast::isa_tree<ast::RescueCase>(tree) ||
+                ast::isa_tree<ast::InsSeq>(tree)) {
+                return;
+            }
+
+            if (auto classDef = ast::cast_tree<ast::ClassDef>(tree)) {
+                if (ast::isa_tree<ast::EmptyTree>(classDef->name)) {
```

Comment:
Why are we looking for an empty name instead of a `classDef->symbol == core::Symbols::root()`?

Reply:
No particular reason, I'll change it to that.